### ISSUE TITLE
Difficult to debug provision error when login is used

### ISSUE THREAD

**Issue by lukaszachy** [teemtee/tmt#1038]
Plan contains invalid image, but when `tmt run -a login` is run, the only failure visible is 
```
stderr (1/1 lines)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ssh: Could not resolve hostname none: Name or service not known

Failed to run command 'ssh -oForwardX11=no -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -S /run/user/17062/tmt/tmpv3ha_5qr root@None export TMT_PLAN_DATA=/var/tmp/tmt/run-109/repro/data; rsync --version --quiet || if [[ ! -f /usr/bin/rpm-ostree ]]; then yum install -y rsync; else yum install -y --installroot=/root/pkg --releasever / rsync && ln -sf /root/pkg/bin/rsync /usr/local/bin/rsync; fi'. Reason: Command returned '255'.
```

Without `login` tmt would print `Could not map 'foooooo' to compose.` But this message is lost forever, even from `log.txt`

IMHO we have two issues:
1. Do not loose error message from log.txt
2. Don't run login if provision failed

```yaml
execute:
    how: tmt
test: echo
discover:
    how: fmf
provision:
    how: virtual
    image: FOOOOOO
```

**Comment by idorax** [comment]
To abort the `login` process, we should tear down `steps.Login` from the `Finish.phases()` if image is invalid after invoking `GuestTestcloud._guess_image_url()`.  

* https://github.com/teemtee/tmt/blob/main/tmt/steps/finish/__init__.py#L71
  ```python
            for phase in self.phases():
                phase.go(guest_copy)
  ```

When `self.phases()` is clear (because the image is invalid), `phase.go(guest_copy)` will never be called. Hence the `login` action can be aborted as expected.

**Comment by lukaszachy** [comment]
@idorax thanks for looking into this. Another `raise ProvisionError` points might be problematic too. I've tried to hit https://github.com/teemtee/tmt/blob/main/tmt/steps/provision/testcloud.py#L512 and it really allows login and original exception is lost forever. (eg. `tmt run -a provision -h virtual -i file:///rubbish login`)

I'm not sure where login should be handled to make it fixed for all provisioners, however I think that problem lies around https://github.com/teemtee/tmt/blob/main/tmt/steps/provision/__init__.py#L137

Maybe log should happen here (to store any ProvisionError). I guess the idea for `finally` is to store other guests (e.g. if multiple are provisioned we need to clean them in finish, right? @psss) however `phase.guest()` shouldn't be true for invalid guest right? (need to properly implement __bool__ in Guest classes)

**Comment by idorax** [comment]
@lukaszachy, thanks for the suggestion! I have proposed a fix, please help to review PR#1322.
* https://github.com/teemtee/tmt/pull/1322
